### PR TITLE
fix: [OIP-442]: Log load test analysis

### PR DESCRIPTION
**Pull request by naveen-harness**
## Describe your changes
Two bugs in load test analysis are fixed as part of this change: 
* "hosts" parameter in log data url can be empty as well as null, if "hosts" parameter is null that means we have to fetch data for all the hosts, while if it is not null and empty that means there are no hosts available. For load test anlaysis, the host in the test data url was being set as not null and empty list, that's why therefore LE wasn't receiving any test data. 
* If baselineVerificationJobInstanceId is null and VerificationJobType is TEST, the analysis state should not transition to log feedback state. 

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [x] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/48566)

### CODE REVIEW

**Comment by naveen-harness** [comment]
trigger uts

**Comment by naveen-harness** [comment]
trigger uts

File: srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/analysis/services/impl/LogAnalysisServiceImpl.java [hunk 1]
```diff
@@ -349,7 +339,7 @@ private String getTestDataUrlForDeploymentLog(AnalysisInput input) {
 
     if (isNotEmpty(input.getTestHosts())) {
       uriBuilder.addParameter("hosts", String.join(",", input.getTestHosts()));
-    } else {
+    } else if (input.getTestHosts() != null) {
```
Comment (AbhijithHarness) [inline srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/analysis/services/impl/LogAnalysisServiceImpl.java:342]: nit: Why is this? we had a not empty check before.... and null check after, do we even need this?

Comment (naveen-harness) [reply srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/analysis/services/impl/LogAnalysisServiceImpl.java:342]: "hosts" parameter in log data url can be empty as well as null, if "hosts" parameter is null that means we have to fetch data for all the hosts, while if it is null that means there are no hosts available. For load test anlaysis, the host in the test data url was being set as an empty list, that's why therefore LE wasn't receiving any test data.

**Review by dhruv-harness — DISMISSED** [review]
Lets add a UT for when hosts are empty, not null

**Comment by naveen-harness** [comment]
trigger uts

**Pull request merged**